### PR TITLE
Add support for secret header validation

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Check out the new https://core.telegram.org/bots/api#setwebhook

### CODE REVIEW

File: src/convenience/frameworks.deno.ts [hunk 1]
```diff
@@ -44,6 +61,13 @@ const serveHttp = (requestEvent: Deno.RequestEvent) => ({
         requestEvent.respondWith(
             new Response(JSON.stringify(json), { status: 200 }),
         ),
+    unauthorized: () =>
+        requestEvent.respondWith(
+            new Response('"unauthorized"', {
+                status: 401,
+                statusText: "secret token is wrong",
```
Comment (Loskir) [inline src/convenience/frameworks.deno.ts:68]: I'm not sure if statusText is meant for that. We should probably put that text into the first argument

File: src/convenience/webhook.ts [hunk 2]
```diff
@@ -91,7 +136,12 @@ export function webhookCallback<C extends Context = Context>(
             await bot.init();
             initialized = true;
         }
-        const { update, respond, end, handlerReturn } = server(...args);
+        const { update, respond, unauthorized, end, handlerReturn, header } =
+            server(...args);
+        if (header !== token) {
```
Comment (Loskir) [inline src/convenience/webhook.ts:141]: shouldn't we rather use `token && header !== token`? to have a way to completely disable this check

File: src/convenience/frameworks.node.ts [hunk 3]
```diff
@@ -10,52 +12,68 @@ const http = (req: IncomingMessage, res: ServerResponse) => ({
...
+    unauthorized: () => res.send(401, "secret token is wrong"),
 });
 
 /** AWS lambda serverless functions */
 const awsLambda = (event: any, _context: any, callback: any) => ({
     update: JSON.parse(event.body),
+    header: event.headers[SECRET_HEADER],
     end: () => callback(null, { statusCode: 200 }),
     respond: (json: string) => callback(null, { statusCode: 200, body: json }),
+    unauthorized: () => callback(null, { statusCode: 401 }),
```
Comment (roj1512) [inline src/convenience/frameworks.node.ts:76]: No body?

Comment (KnorpelSenf) [reply src/convenience/frameworks.node.ts:76]: https://t.me/grammyjs/58046

File: src/convenience/webhook.ts (hunk 2 above)
Comment (KnorpelSenf) [reply src/convenience/webhook.ts:141]: If Telegram delivers a token, but it is not supplied here, then there is a mismatch and the check will fail. Currently, the two configurations must match exactly.

What you are suggesting is to ignore if Telegram sends a token, is that better?

**Comment by KnorpelSenf** [comment]
I have tested this with a bot and it works, but we are going to perform a follow-up PQ by @kraftwerk28 which will align the different frameworks. I suggest to merge and release as soon as https://github.com/grammyjs/grammY/pull/230#discussion_r902485165 is sorted out.

**Pull request merged**